iothub/device/tests/ClientPropertyCollectionTestsNewtonsoft.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Devices.Client;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class ClientPropertyCollectionTestsNewtonsoft
+    {
+        private const string propertyName = "propertyOne";
+        private const string simplePropertyValue = "propertyValue";
+        private const string componentName = "testableComponent";
+        private const string writablePropertyDescription = "testableWritablePropertyDescription";
+        private const string updatedPropertyValue = "updatedPropertyValue";
+        
+        private const string twinJson = "{ \"propertyOne\" : \"propertyValue\" }";
+        private const string twinJsonWritableProperty = "{ \"propertyOne\" : { \"value\" : \"propertyValue\",  \"ac\" : 200, \"av\" : 2, \"ad\" : \"testableWritablePropertyDescription\" } }";
+        private const string twinJsonWithComponent = "{ \"testableComponent\" : { \"__t\" : \"c\",  \"propertyOne\" : \"propertyValue\" } }";
+        private const string twinJsonWritablePropertyWithComponent = "{ \"testableComponent\" : { \"__t\" : \"c\", \"propertyOne\" : { \"value\" : \"propertyValue\",  \"ac\" : 200, \"av\" : 2, \"ad\" : \"testableWritablePropertyDescription\" } } }";
+
+        private static TwinCollection collectionToRoundTrip = new TwinCollection(twinJson);
+        private static TwinCollection collectionWritablePropertyToRoundTrip = new TwinCollection(twinJsonWritableProperty);
+        private static TwinCollection collectionWithComponentToRoundTrip = new TwinCollection(twinJsonWithComponent);
+        private static TwinCollection collectionWritablePropertyWithComponentToRoundTrip = new TwinCollection(twinJsonWritablePropertyWithComponent);
+
+        [TestMethod]
+        public void ClientPropertyCollectionNewtonsoft_CanGetSimpleValue()
```

Comment:
this should also cover the range of supported data types

Reply:
got it so all DTDL primitives